presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/PlanMatchPattern.java

```diff
@@ -247,9 +247,9 @@ public static PlanMatchPattern sort(PlanMatchPattern source)
         return node(SortNode.class, source);
     }
 
-    public static PlanMatchPattern topN(long count, List<String> orderBy, PlanMatchPattern source)
+    public static PlanMatchPattern topN(long count, ImmutableMap<String, SortOrder> orderingScheme, PlanMatchPattern source)
```

Comment:
`Map`

also, maybe leave original method, the `topN(long count, List<String> orderBy, PlanMatchPattern source)` signature looks like more convenient

Reply:
I think pure `Map` is confusing in this case, because `ImmutableMap` is guaranteed to preserve key insertion order (when iterating). What do you think?

> also, maybe leave original method, the topN(long count, List<String> orderBy, PlanMatchPattern source) signature looks like more convenient

but it implicitly assumes some order. I will leave modified method for now.

Reply:
> ImmutableMap is guaranteed to preserve key insertion order (when iterating).

if you're after something that maintains an insertion ordering, use `List`. I know `ImmutableMap` does that too, but making implicit use of that property is an API-obfuscation.

What about:
1. having method that takes 1 symbol and 1 SortOrder
2. having an overload that takes (1 symbol, 1 SortOrder, 2nd symbol, 2nd SortOrder) (if this is still insufficient for test purposes, introduce a builder)

Reply:
IMO this change is unrelated to the refactoring you are doing, I would revert that.